community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py

```diff
@@ -99,19 +99,32 @@ def instance_properties(nodeset, model, placement_group, labels=None):
         props.scheduling.maintenanceInterval = nodeset.maintenance_interval
 
     if nodeset.dws_flex.enabled:
-        update_props_dws(props,nodeset.dws_flex)
+        update_props_dws(props, nodeset.dws_flex, job_id)
 
     # Override with properties explicit specified in the nodeset
     props.update(nodeset.get("instance_properties") or {})
     
     return props
 
-def update_props_dws(props:dict,dws_flex:dict) -> None:
+def update_props_dws(props:object, dws_flex:object, job_id: int) -> None:
     props.scheduling.onHostMaintenance = "TERMINATE"
     props.scheduling.instanceTerminationAction = "DELETE"
-    props.scheduling.maxRunDuration['seconds'] = dws_flex.max_run_duration
     props.reservationAffinity['consumeReservationType'] = "NO_RESERVATION"
 
+    if dws_flex.use_job_duration:
+        current_job = lookup().job(job_id)
+        if not current_job or not current_job.duration:
+            props.scheduling.maxRunDuration['seconds'] = dws_flex.max_run_duration
+            return
```

Comment:
This return-termination doesn't looks good given that function does more than just figuring out value of `duration`.
Consider moving it into separate function dedicated to duration value, or rewrite so you don't need to use return-termination. E.g:

```py
dur = timedelta(seconds=dws_flex.max_run_duration)
if dws_flex.use_job_duration and job_id and (job := lookup().job(job_id)) and job.duration:
  min_dur, max_dur = timedelta(seconds=30), timedelta(weeks=2)
  dur = min(max(job.duration, min_dur), max_dur)
props.scheduling.maxRunDuration['seconds'] = int(current_job.duration.total_seconds())
```

Reply:
Sounds good, kind've did both since it seemed to make things a bit more readable. Not sure if its considered a code-smell or not to pass parameters through functions that don't make use of it but open to suggestions.